pkg/cloud/azure/actuators/cluster/reconciler.go

```diff
@@ -17,49 +17,259 @@ limitations under the License.
 package cluster
 
 import (
+	"fmt"
+	"hash/fnv"
+	"strings"
+
 	"github.com/pkg/errors"
 	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/actuators"
 	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services"
 	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/certificates"
 	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/groups"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/internalloadbalancers"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/publicips"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/publicloadbalancers"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/routetables"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/securitygroups"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/subnets"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services/virtualnetworks"
 )
 
 // Reconciler are list of services required by cluster actuator, easy to create a fake
 type Reconciler struct {
-	scope           *actuators.Scope
-	groupsSvc       services.Service
-	certificatesSvc services.Service
+	scope            *actuators.Scope
+	certificatesSvc  services.Service
+	groupsSvc        services.Service
+	vnetSvc          services.Service
+	securityGroupSvc services.Service
+	routeTableSvc    services.Service
+	subnetsSvc       services.Service
```

Comment:
`services.Service` is not very clear. Surely we can do better with idiomatic package naming?  One of the primary recommendations of Go best practices is to make sure that there aren't redundancies/ambiguities when specifying the `package` + `struct` name.

Reply:
ugh! me too, i hate services.Service. Ideally i prefer azure.Service and all the services defined in sub package. Is it ok to rename this in next PR?

Reply:
Would prefer to avoid the stuttering as well. We need to fix the rename in this PR, before the convention sticks.
How about `azure.Service` or `provider.Service` (if we want to be more generic)?

Reply:
Yep updated, thanks :) renamed to azure.Service